### PR TITLE
Go install all components

### PR DESCRIPTION
**Pull request by jimmykarily**

### CODE REVIEW

File: cli/helpers/exec.go
Comment (andreas-kupries) [inline cli/helpers/exec.go:110]: Does it make sense to move this statement to just before `s.Start()` above ?

In the position it is in it feels as if the main goroutine accesses the data structures of the goroutine inside of the spinner, i.e. changing the suffix underneath it. Before `s.Start()` it would make the change before the spinner's goroutine is active.

Hm ... Will activate the `-race` option for building he app, see what the data race detectors tells me.

Comment (andreas-kupries) [reply cli/helpers/exec.go:110]: Yup. Immediate data race involving the spinner.
Now to check if my proposed change will fix it.

```
ARNING: DATA RACE
Read at 0x00c00020a718 by goroutine 27:
  github.com/briandowns/spinner.(*Spinner).Start.func1()
      /home/work/SUSE/go/pkg/mod/github.com/briandowns/spinner@v1.12.0/spinner.go:307 +0x339

Previous write at 0x00c00020a718 by main goroutine:
  github.com/suse/carrier/cli/kubernetes.(*Cluster).WaitForPodBySelectorRunning()
      /home/work/SUSE/dev/carrier/cli/kubernetes/cluster.go:178 +0x229
  github.com/suse/carrier/cli/deployments.Quarks.apply()
      /home/work/SUSE/dev/carrier/cli/deployments/quarks.go:70 +0x429
  github.com/suse/carrier/cli/deployments.Quarks.Deploy()
      /home/work/SUSE/dev/carrier/cli/deployments/quarks.go:95 +0x2f5
  github.com/suse/carrier/cli/deployments.(*Quarks).Deploy()
      <autogenerated>:1 +0xd7
  github.com/suse/carrier/cli/kubernetes.(*Installer).Install()
      /home/work/SUSE/dev/carrier/cli/kubernetes/installer.go:96 +0x1d8
  github.com/suse/carrier/cli/cmd.Install()
      /home/work/SUSE/dev/carrier/cli/cmd/install.go:103 +0x38a
  github.com/spf13/cobra.(*Command).execute()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x8e0
  github.com/spf13/cobra.(*Command).ExecuteC()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x499
  github.com/suse/carrier/cli/cmd.Execute()
      /home/work/SUSE/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895 +0x2c5
  main.main()
      /home/work/SUSE/dev/carrier/cli/main.go:6 +0x2f

Goroutine 27 (running) created at:
  github.com/briandowns/spinner.(*Spinner).Start()
      /home/work/SUSE/go/pkg/mod/github.com/briandowns/spinner@v1.12.0/spinner.go:281 +0xef
  github.com/suse/carrier/cli/kubernetes.(*Cluster).WaitForPodBySelectorRunning()
      /home/work/SUSE/dev/carrier/cli/kubernetes/cluster.go:176 +0xee
[...]
```

Comment (andreas-kupries) [reply cli/helpers/exec.go:110]: No, apparently not. Hm. That means the race seen above is not what I thought. Looking deeper now instead of assuming.

Comment (andreas-kupries) [reply cli/helpers/exec.go:110]: Ah! The spinner creations in the cluster.go have the same ordering issue on `.Suffix`. 'k, change these too.

Comment (andreas-kupries) [reply cli/helpers/exec.go:110]: The one in the loop over the pods (in `WaitForPodBySelectorRunning`) requires a full cycle of "stop; modify; start". ... Pity that the spinner package does not have a method change the suffix mid operation. That could have used the internal lock employed by the spinner.

And with that no races in this area are seen any longer.
Will make a PR for this now.

Comment (andreas-kupries) [reply cli/helpers/exec.go:110]: PR #44

Comment (andreas-kupries) [reply cli/helpers/exec.go]: Merged. Resolving.

Comment (jimmykarily) [reply cli/helpers/exec.go]: Merged, thanks!

**Comment by andreas-kupries** [comment]
I see issues when deploying, where the eirini api, task reporter, and metrics are in CLBO. All report issues accessing cert files in their kube logs:

```
work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-api
fetching pods => 25
kubectl logs eirini-api-7bf8f85cc9-845tg --namespace eirini-core

W0115 08:48:13.884190       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
Failed to build TLS config: failed to read certificate(s) at path "/etc/eirini/certs/ca.crt": open /etc/eirini/certs/ca.crt: no such file or directory
{"timestamp":"2021-01-15T08:48:13.885120879Z","level":"info","source":"handler","message":"handler.opi-connected","data":{}}


work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-task
fetching pods => 25
kubectl logs eirini-task-reporter-96bf7c555-27scg --namespace eirini-core

W0115 08:48:23.887806       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0115 08:48:23.888611       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
an unexpected error occurred: "CC Cert" file at "/etc/cf-api/certs/tls.crt" does not exist


work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-metrics
fetching pods => 25
kubectl logs eirini-metrics-6cc65d55db-d4w4d --namespace eirini-core

W0115 08:48:26.887957       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0115 08:48:26.888870       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
an unexpected error occurred: "Loggregator CA" file at "/etc/loggregator/certs/ca.crt" does not exist
```

**Comment by jimmykarily** [comment]
> I see issues when deploying, where the eirini api, task reporter, and metrics are in CLBO. All report issues accessing cert files in their kube logs:
> 
> ```
> work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-api
> fetching pods => 25
> kubectl logs eirini-api-7bf8f85cc9-845tg --namespace eirini-core
> 
> W0115 08:48:13.884190       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> Failed to build TLS config: failed to read certificate(s) at path "/etc/eirini/certs/ca.crt": open /etc/eirini/certs/ca.crt: no such file or directory
> {"timestamp":"2021-01-15T08:48:13.885120879Z","level":"info","source":"handler","message":"handler.opi-connected","data":{}}
> 
> 
> work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-task
> fetching pods => 25
> kubectl logs eirini-task-reporter-96bf7c555-27scg --namespace eirini-core
> 
> W0115 08:48:23.887806       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> W0115 08:48:23.888611       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> an unexpected error occurred: "CC Cert" file at "/etc/cf-api/certs/tls.crt" does not exist
> 
> 
> work@tagetarl:~/SUSE/dev/carrier> k logs :eirini-metrics
> fetching pods => 25
> kubectl logs eirini-metrics-6cc65d55db-d4w4d --namespace eirini-core
> 
> W0115 08:48:26.887957       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> W0115 08:48:26.888870       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
> an unexpected error occurred: "Loggregator CA" file at "/etc/loggregator/certs/ca.crt" does not exist
> ```

Fixed with this: https://github.com/SUSE/carrier/pull/35/commits/2fb70d22ef5a0353c95a754f856436a773e34959

File: cli/deployments/eirini.go
Comment (jimmykarily) [inline cli/deployments/eirini.go:188]: Dummy copy-paste code. Upgrading/Deleting/etc need more sophisticated handling. That's our of scope for this PR.

File: cli/deployments/tekton.go
Comment (jimmykarily) [inline cli/deployments/tekton.go:221]: These methods are not properly "scoped" I think. The are available all over the `deployments` package although they are only relevant to this deployment. Maybe they should have this deployment as a receiver.

File: cli/embedded-files/eirini/quarks-secrets.yaml
Comment (jimmykarily) [inline cli/embedded-files/eirini/quarks-secrets.yaml:1]: This file achieves the same results as this code: https://github.com/SUSE/carrier/blob/main/eirini/eirini-release/deploy/scripts/generate_eirini_tls.sh#L14-L23

File: cli/embedded-files/tekton/ingress.yaml
Comment (jimmykarily) [inline cli/embedded-files/tekton/ingress.yaml:11]: Missed this. the Tekton deployment needs a bit more code to handle this: https://github.com/SUSE/carrier/blob/main/tekton/install#L61-L63

File: cli/helpers/downloader.go
Comment (jimmykarily) [inline cli/helpers/downloader.go:1]: This was used to download the eirini release but we changed that to be an embedded tarball. I kept this around because we could use it to fetch our own dependencies if needed (e.g. download a specific version of `kubectl` and `helm` if they don't exist locally and after asking the user).

File: cli/embedded-files/tekton/ingress.yaml
Comment (jimmykarily) [reply cli/embedded-files/tekton/ingress.yaml]: Added here: https://github.com/SUSE/carrier/pull/35/commits/22bf48d95a02d62d2ef988e4cd2918f298c3b7a0

File: cli/statik/statik.go
Comment (andreas-kupries) [inline cli/statik/statik.go]: Nit: Indent is incorrect. Just run through `make fmt` to fix.

File: cli/deployments/eirini.go
Comment (andreas-kupries) [inline cli/deployments/eirini.go:21]: Nit: Remove empty line ?

File: cli/deployments/eirini.go
Comment (andreas-kupries) [inline cli/deployments/eirini.go:253]: Nit: Inconsistent indent with the imagePullSecrets.

File: cli/deployments/tekton.go
Comment (andreas-kupries) [inline cli/deployments/tekton.go:23]: Nit: Remove empty line ?!

File: cli/deployments/tekton.go
Comment (andreas-kupries) [inline cli/deployments/tekton.go:200]: Nit: Empty line.

File: cli/deployments/eirini.go
Comment (andreas-kupries) [inline cli/deployments/eirini.go:160]: Nit: Empty line

File: cli/deployments/registry.go
Comment (andreas-kupries) [inline cli/deployments/registry.go:93]: Nit: empty line

File: cli/helpers/embedded_files.go
Comment (andreas-kupries) [inline cli/helpers/embedded_files.go:18]: Nit: Better name maybe `ExtractFile`  ?

**Review by andreas-kupries — APPROVED** [review]
Saw only nits, nothing blocking the merge.

Note however, after merge `openssl` should be added to the set of dependencies check on cli start  (currently only helm and kubectl).

File: cli/deployments/eirini.go
Comment (jimmykarily) [reply cli/deployments/eirini.go:160]: A habit I got from [this linter](https://github.com/ssgreg/nlreturn). It's also a convention in other languages to make the return statement more visible. I would like to keep it, hoping in the future to add some linters to our project (a list: https://golangci-lint.run/usage/linters/). Is that ok?

File: cli/deployments/eirini.go
Comment (jimmykarily) [reply cli/deployments/eirini.go:253]: They look the same in my vim (but one is spaces the other is tab). Fixing.

File: cli/deployments/registry.go
Comment (jimmykarily) [reply cli/deployments/registry.go:93]: see my reply on the other comment

File: cli/deployments/eirini.go
Comment (andreas-kupries) [reply cli/deployments/eirini.go:159]: Sure. As I said, nits, mostly style. I am able to cope.

File: cli/deployments/eirini.go
Comment (andreas-kupries) [reply cli/deployments/eirini.go:252]: Ah. Yes tab vs spaces is seen easily in various diff displays.

**Pull request merged**
